File: Legacy/PathFinderD/Robotics3/Cibertools/jClient/Engine/DodgeObstacle.java
package Engine;

import static Engine.Sensors.*;
import ciberIF.ciberIF;

public class DodgeObstacle implements Behaviour {

    @Override
    public void execute(ciberIF cif) {
        /*boolean avoider = avoidRepetitions();
        if(avoider)
            System.out.println("AVOID!");
        */
        // Rotate right on obstacles close in front or close on the left
        if( irSensor1 > irSensor2 ) { // avoider ||
    	    cif.DriveMotors(0.08,-0.08);
        }
        // Rotate left in obstacles close on the right
        else {
            cif.DriveMotors(-0.08,0.08);
        }
        
    }

    @Override
    public boolean isPossible() {
        return (irSensor0 > 3.0 || irSensor1 > 3.0 || irSensor2 > 3.0);     //there's an obstacle somewhere at front/right/left
    }

    @Override
    public String getName() {
        return "DodgeObstacle";
    }
    
}

File: MazeRunner/DodgeObstacle.c
#include "Sensors.h"
#include "rmi-mr32.h"
#include <math.h>
#include <string.h>

#include "DodgeObstacle.h"

char* do_getName() {
    return "DodgeObstacle";
}

void do_execute() {

    /* Rotate right on obstacles close in front or close on the left */
    int mod = 1;
    
    if(obstValLeft > obstValRight){ 
		mod=-1;
    }

    if(obstValFront < 15)			// if obstacle in front, rotate quickly
    	setVel2(mod*40, mod*-40);	
    else							// if obstacle on the sides, rotate slowly
    	setVel2(mod*20, mod*-20);
}

bool do_isPossible() {
    /* There's an obstacle somewhere at front/right/left */
    return obstValFront < 15 || obstValLeft < 10 || obstValRight < 10;
}

